### DISCUSSION TITLE
set result labels

### DISCUSSION THREAD

**Discussion by Sobes76rus** [taskiq-python/taskiq#155] [Ideas]
Hello there! I would like to provide some labels to `set_result` method. 
For example i wanna control `ttl` for each task.
Can we do it?

**Comment by s3rius** [comment]
Actually it's a good idea.  But currently there's no way to do so. I've created an issue for that.

**Reply by s3rius** [reply]
The issue is #156.

**Reply by s3rius** [reply]
Guess the best way to do so is to add labels to TaskiqResult model.

**Reply by Sobes76rus** [reply]
As a workaround, we can raise a custom exception like `NoResultError` with the task result data and process it in `resultbackend`.

**Reply by s3rius** [reply]
Hi, @Sobes76rus. With the latest update you can easily inherit from any result backend an change it's logic based on labels. 

Here's an example with RedisResultBackend.


```python
import pickle
from redis.asyncio import Redis
from taskiq.result import TaskiqResult
from taskiq_redis import RedisAsyncResultBackend


class MyResultBackend(RedisAsyncResultBackend):
    async def set_result(
        self,
        task_id: str,
        result: TaskiqResult,
    ) -> None:
        ttl = result.labels.get("result_ttl")

        redis_set_params = {
            "name": task_id,
            "value": pickle.dumps(result),
        }
        if ttl or self.result_ex_time:
            redis_set_params["ex"] = ttl or self.result_ex_time
        elif self.result_px_time:
            redis_set_params["px"] = self.result_px_time

        async with Redis(connection_pool=self.redis_pool) as redis:
            await redis.set(**redis_set_params)

```

To set TTL for a task run simply pass it as label.

```python
    await my_task.kicker().with_labels(result_ttl=10).kiq()
```

Or you can even define per-task TTL.

```python
@broker.task(result_ttl=10)
async def my_task():
    return 1
```

This functionality available starting from version https://github.com/taskiq-python/taskiq/releases/tag/0.7.2.

**Reply by Sobes76rus** [reply]
Nice! Thank you =)
